CRM/Utils/Array.php

```diff
@@ -1211,6 +1211,35 @@ public static function pathSync(&$params, $canonPath, $altPath, ?callable $filte
     }
   }
 
+  /**
+   * Take one well-defined item out of a single-item list.
+   *
+   * Assert that the list genuinely contains *exactly* one item.
+   *
+   * @param iterable $items
+   * @param string $recordTypeLabel
+   * @return mixed
+   *   The first (and only) item from the $items list.
+   * @throws \CRM_Core_Exception
+   */
+  public static function single(iterable $items, string $recordTypeLabel = 'record') {
```

Comment:
IMO the second param adds more complexity than benefit; it would be fine to just call everything a "record".